content/hardware/08.edu/solution-and-kits/alvik/tutorials/user-manual/user-manual.md

```diff
@@ -240,6 +346,1018 @@ Open **Arduino Lab for MicroPython** and **connect** Alvik. Then:
+
+
+### Reading Buttons
+
+The Arduino Alvik robot is equipped with several touch buttons that can be used for various input purposes. The following functions are available to read the state of each button:
+
+- Detect **any** button pressed.
+
+   The `[get_touch_any](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_any)` function returns true if any of the buttons is pressed.
+
+
+- Detect the **OK** button pressed.
+
+   The `[get_touch_ok](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_ok)` function returns true if the OK button is pressed.
+
+
+- Detect the **Cancel** button pressed.
+
+   The `[get_touch_cancel](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_cancel)` function returns true if the Cancel button is pressed.
+
+
+- Detect the **Center** button pressed.
+
+   The `[get_touch_center](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_center)` function returns true if the Center button is pressed.
+
+
+- Detect the **Up** button pressed.
+
+   The `[get_touch_up](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_up)` function returns true if the Up button is pressed.
+
+
+- Detect the **Left** button pressed.
+
+   The `[get_touch_left](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_left)` function returns true if the Left button is pressed.
+
+
+- Detect the **Down** button pressed.
+
+   The `[get_touch_down](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_down)` function returns true if the Down button is pressed.
+
+
+- Detect the **Right** button pressed.
+
+   The `[get_touch_right](https://docs.arduino.cc/tutorials/alvik/api-overview/#get_touch_right)` function returns true if the Right button is pressed.
+
+
+**Example**
+The following example demonstrates how to read the state of each button and print its name when pressed:
+
+```python
+from arduino_alvik import ArduinoAlvik
+from time import sleep_ms
+import sys
+
+alvik = ArduinoAlvik()
+alvik.begin()
+
+while True:
+    try:
+        if alvik.get_touch_any():
+            if alvik.get_touch_up():
+                print("UP")
+            if alvik.get_touch_down():
+                print("DOWN")
+            if alvik.get_touch_left():
+                print("LEFT")
+            if alvik.get_touch_right():
+                print("RIGHT")
+            if alvik.get_touch_ok():
+                print("OK")
+            if alvik.get_touch_cancel():
+                print("CANCEL")
+            if alvik.get_touch_center():
+                print("CENTER")
+
+        sleep_ms(100)
+    except KeyboardInterrupt as e:
+        print('over')
+        alvik.stop()
+        sys.exit()
+```
+
+This example provides a simple way to read and respond to button presses on the Arduino Alvik robot.
+Feel free to expand to it by adding customized responses to each different press.
+
+
+### Detecting Obstacles

```

Comment:
I would move this section above to the Out of the box experience and explain here how to program the ToF same as we are doing with motors and buttons.

Reply:
yeap